src/Uno.UWP/Globalization/NumberFormatting/PermilleFormatter.cs

```diff
@@ -1,99 +1,100 @@
+	public string ResolvedLanguage => _translator.ResolvedLanguage;
 
-		public string ResolvedLanguage => _translator.ResolvedLanguage;
+	public int FractionDigits { get => _formatterHelper.FractionDigits; set => _formatterHelper.FractionDigits = value; }
 
-		public int FractionDigits { get => _formatterHelper.FractionDigits; set => _formatterHelper.FractionDigits = value; }
+	public INumberRounder? NumberRounder { get; set; }
 
-		public INumberRounder NumberRounder { get; set; }
+	public bool IsZeroSigned { get => _formatterHelper.IsZeroSigned; set => _formatterHelper.IsZeroSigned = value; }
 
-		public bool IsZeroSigned { get => _formatterHelper.IsZeroSigned; set => _formatterHelper.IsZeroSigned = value; }
+	public int SignificantDigits { get => _formatterHelper.SignificantDigits; set => _formatterHelper.SignificantDigits = value; }
 
-		public int SignificantDigits { get => _formatterHelper.SignificantDigits; set => _formatterHelper.SignificantDigits = value; }
+	public string Format(double value) => FormatDouble(value);
 
-		public string Format(double value) => FormatDouble(value);
+	public string FormatDouble(double value)
+	{
+		if (!_formatterHelper.TryValidate(value, out string text))
+		{
+			return text;
+		}
 
-		public string FormatDouble(double value)
+		if (NumberRounder != null)
 		{
-			if (!_formatterHelper.TryValidate(value, out string text))
-			{
-				return text;
-			}
-
-			if (NumberRounder != null)
-			{
-				value = NumberRounder.RoundDouble(value);
-			}
-
-			string formatted = string.Empty;
-
-			if (value == 0d)
-			{
-				formatted = _formatterHelper.FormatZero(value);
-			}
-			else
-			{
-				// due to accuracy precision "MultiplyByPow10" is used for multiplication
-				value = value.MultiplyByPow10(Pow10);
-				formatted = _formatterHelper.FormatDoubleCore(value);
-			}
-
-			formatted = _translator.TranslateNumerals($"{formatted}{_symbol}");
-			return formatted;
+			value = NumberRounder.RoundDouble(value);
 		}
 
-		public double? ParseDouble(string text)
+		string formatted = string.Empty;
+
+		if (value == 0d)
 		{
-			text = _translator.TranslateBackNumerals(text);
+			formatted = _formatterHelper.FormatZero(value);
+		}
+		else
+		{
+			// due to accuracy precision "MultiplyByPow10" is used for multiplication
+			value = value.MultiplyByPow10(Pow10);
+			formatted = _formatterHelper.FormatDoubleCore(value);
+		}
 
-			if (!text.EndsWith(_symbol, StringComparison.Ordinal))
-			{
-				return null;
-			}
+		formatted = _translator.TranslateNumerals($"{formatted}{_symbol}");
+		return formatted;
+	}
 
-			text = text.Substring(0, text.Length - _symbol.Length);
+	public double? ParseDouble(string text)
+	{
+		text = _translator.TranslateBackNumerals(text);
+
+		if (!text.EndsWith(_symbol, StringComparison.Ordinal))
+		{
+			return null;
+		}
 
-			var result = _formatterHelper.ParseDoubleCore(text);
+		text = text.Substring(0, text.Length - _symbol.Length);
```

Comment:
Like above

Reply:
@workgroupengineering Is it necessary to use `StringBuilder` here because there is just one line that can be converted to StringBuilder style so if I use StringBuilder the `ToString()` creates new string?

Reply:
you can `text.AsSpan().Slice(0,text.Length - _symbol.Length)`;

Reply:
I would rather use `StringBuilder` as you said first.